src/Datadog.Trace/Tagging/RabbitMQTags.cs

```diff
@@ -0,0 +1,47 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Datadog.Trace.ExtensionMethods;
+
+namespace Datadog.Trace.Tagging
+{
+    internal class RabbitMQTags : InstrumentationTags
+    {
+        protected static readonly IProperty<string>[] RabbitMQTagsProperties =
+            InstrumentationTagsProperties.Concat(
+                new Property<RabbitMQTags, string>(Trace.Tags.InstrumentationName, t => t.InstrumentationName, (t, v) => t.InstrumentationName = v),
+                new Property<RabbitMQTags, string>(Trace.Tags.AmqpCommand, t => t.Command, (t, v) => t.Command = v),
+                new Property<RabbitMQTags, string>(Trace.Tags.AmqpDeliveryMode, t => t.DeliveryMode, (t, v) => t.DeliveryMode = v),
+                new Property<RabbitMQTags, string>(Trace.Tags.AmqpExchange, t => t.Exchange, (t, v) => t.Exchange = v),
+                new Property<RabbitMQTags, string>(Trace.Tags.AmqpQueue, t => t.Queue, (t, v) => t.Queue = v),
+                new Property<RabbitMQTags, string>(Trace.Tags.AmqpRoutingKey, t => t.RoutingKey, (t, v) => t.RoutingKey = v),
+                new Property<RabbitMQTags, string>(Trace.Tags.MessageSize, t => t.MessageSize, (t, v) => t.MessageSize = v));
+
+        private string _spanKind = SpanKinds.Client;
+
+        public override string SpanKind => _spanKind;
+
+        public string InstrumentationName { get; set; }
+
+        public string Command { get; set; }
+
+        public string DeliveryMode { get; set; }
+
+        public string Exchange { get; set; }
+
+        public string RoutingKey { get; set; }
+
+        public string MessageSize { get; set; }
+
+        public string Queue { get; set; }
+
+        public void SetSpanKind(string spanKind)
```

Comment:
It looks like `InstrumentationTags.SpanKind` was designed to be readonly. Is this going to cause any issues? Instead of this workaround, should we pass the value into `RabbitMQTags`'s ctor? Or make `SpanKind` not readonly? cc @kevingosse

Reply:
I believe the only consequence of SpanKind being readonly is that `SetTag(Tags.SpanKind, value)` will throw an exception. Setting it through `SetSpanKind` should be fine.

Oh, and this may cause the TagsListTests.CheckProperties unit test to fail 🤔

Reply:
> `SetTag(Tags.SpanKind, value)` will throw an exception

When using manual instrumentation, this is how users set this tag, so we need to fix.

Reply:
@kevingosse @lucaspimentel I tried to address this in f45a8554db67ca7b581e4e35e480e9bcb38c857a . Is this solution fine or would you suggest more changes?

Reply:
LGTM, but I'll defer to @kevingosse since he is more familiar with the new tagging framework. Thanks for setting that tag in _all_ integrations, too.

Reply:
> When using manual instrumentation, this is how users set this tag, so we need to fix.

This will throw only for automatic spans, e.g. those that inherits from `InstrumentationTags`.

If possible I'd rather we leave the property as readonly, and set it from the constructor for RabbitMQ tags. Otherwise we increase the size of all those objects (the good thing about the current property is that it's not backed by a field)

Reply:
Customers can get the `Span` from `Tracer.Instance.ActiveScope.Span` and then call `Span.SetTag` on it, so we should avoid throwing exceptions if we can avoid it. Maybe we can make calls on `ReadOnlyProperty` a no-op instead?

Reply:
So what do you think of this set of changes:
- Revert tags changes so `SpanKind` is readonly again
- Keep change where RabbitMQ span kind is passed in constructor
- Change `ReadOnlyProperty` to do a no-op instead of a throw?

Reply:
Sounds good to me 👍

Reply:
Makes sense. Thanks everyone!

Reply:
Thanks for the input everyone. Changes made:
- `SpanKind` changes made in 5b16ce6bcdd702658bf629ad2263c77afe7fa56a
- `ReadOnlyProperty` change made in 4bcd2f8a226ff86457fa55fa1c7236b3602bc7db